name: Test Action 2

# Trigger this workflow either manually or when a new change is pushed
on:
  push:
  workflow_dispatch:

jobs:
  test-octave-action:
    runs-on: ubuntu-latest
    steps:
      - name: Check out repository
        uses: actions/checkout@v4
      - name: Run Octave Unit tests
        uses: ./ # run action.yml from the root of this repo
        # env:
        #   WORKDIR: ./test_cases
        with:
          test_case: MyTestCases

      - name: Run Failing Octave Unit tests
        id: should-fail
        continue-on-error: true
        uses: ./ # run action.yml from the root of this repo
        # env:
        #   WORKDIR: ./test_cases
        with:
          test_case: MyFailingTestCases

      - name: Map unwanted success of 2nd unit tests to error
        if: steps.should-fail.outcome == 'success'
        uses: actions/github-script@v3
        with:
          script: |
            core.setFailed("Failing unit tests should have failed but didn't.")
